File: schemas/video-conferencing/drizzle/meeting_reactions.ts
// meeting_reactions: Ephemeral emoji reactions sent during a meeting.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { meetings } from "./meetings";
import { users } from "./users";

export const meetingReactions = pgTable(
  "meeting_reactions",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    meetingId: uuid("meeting_id").notNull().references(() => meetings.id, { onDelete: "cascade" }),
    userId: uuid("user_id").notNull().references(() => users.id, { onDelete: "cascade" }),
    emoji: text("emoji").notNull(),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_meeting_reactions_meeting_id_created_at").on(table.meetingId, table.createdAt),
    index("idx_meeting_reactions_meeting_id_emoji").on(table.meetingId, table.emoji),
  ]
);

File: schemas/video-conferencing/spacetimedb/meeting_reactions.rs
// meeting_reactions: emoji reactions sent by participants during a meeting.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

// Composite index (meeting_id, created_at) — not supported in SpacetimeDB
// Composite index (meeting_id, emoji) — not supported in SpacetimeDB

#[spacetimedb::table(name = meeting_reactions, public)]
pub struct MeetingReaction {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub meeting_id: String, // UUID — FK → meetings.id (cascade delete)
    pub user_id: String, // UUID — FK → users.id (cascade delete)
    pub emoji: String,
    pub created_at: Timestamp,
}
